### PR TITLE
Use download stats for updating `addon.average_daily_users` for dictionaries and langpacks

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/14609

### CODE REVIEW

File: src/olympia/addons/cron.py [hunk 1]
```diff
@@ -39,6 +39,17 @@ def update_addon_average_daily_users():
...
+        # BigQuery does not have data for add-ons with type other than those in
+        # `ADDON_TYPES_WITH_STATS` so we use download counts instead.
+        # See: https://github.com/mozilla/addons-server/issues/14609
+        d += (
+            Addon.objects
+                .exclude(type__in=amo.ADDON_TYPES_WITH_STATS)
+                .annotate(count=Sum('downloadcount__count'))
+                .values_list('guid', 'count')
+                .order_by('guid')
+                .all()
```
Comment (willdurand) [inline src/olympia/addons/cron.py:51]: @diox I don't like the fact that we retrieve all results first but I don't have a nicer way. WDYT?

Comment (diox) [reply src/olympia/addons/cron.py]: That's probably fine, but I'd do the `Addon.objects.[...]` call first. The reason is, you might lose the connection to the database if the big query call takes too much time, so it'd be safer if the 2 were swapped.

File: src/olympia/addons/cron.py [hunk 2]
```diff
@@ -38,7 +38,18 @@ def update_addon_average_daily_users():
...
-        d = get_addons_and_average_daily_users_from_bigquery()
+        # BigQuery does not have data for add-ons with type other than those in
+        # `ADDON_TYPES_WITH_STATS` so we use download counts instead.
+        # See: https://github.com/mozilla/addons-server/issues/14609
+        d = list(
+            Addon.objects
+            .exclude(type__in=amo.ADDON_TYPES_WITH_STATS)
+            .annotate(count=Sum('downloadcount__count'))
+            .values_list('guid', 'count')
+            .order_by('guid')
```
Comment (eviljeff) [inline src/olympia/addons/cron.py:49]: does the response need to be ordered?  guid ordering is pretty arbitrary.

File: src/olympia/addons/cron.py [hunk 3]
```diff
@@ -38,7 +38,18 @@ def update_addon_average_daily_users():
...
+            .all()
```
Comment (eviljeff) [inline src/olympia/addons/cron.py:50]: don't need the `.all()` after you've done other queryset operations to the model manager.

File: src/olympia/addons/cron.py [hunk 4]
```diff
@@ -38,7 +38,18 @@ def update_addon_average_daily_users():
 
     kwargs = {'id_field': 'pk'}
     if waffle.switch_is_active('use-bigquery-for-addon-adu'):
-        d = get_addons_and_average_daily_users_from_bigquery()
+        # BigQuery does not have data for add-ons with type other than those in
+        # `ADDON_TYPES_WITH_STATS` so we use download counts instead.
+        # See: https://github.com/mozilla/addons-server/issues/14609
+        d = list(
```
Comment (eviljeff) [inline src/olympia/addons/cron.py:44]: Can we take the chance to make a change to have a longer variable name please?  Single letter vars can be problematic.

File: src/olympia/addons/cron.py (hunk 2 above)
Comment (willdurand) [reply src/olympia/addons/cron.py:49]: I think we can get rid of that indeed

**Pull request merged**